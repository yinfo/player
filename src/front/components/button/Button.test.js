// @flow

import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './Button';

describe('Button component', () => {
  const onClickMock = jest.fn();
  const preventDefault = jest.fn();

  it('renders as expected', () => {
    const component = shallow(
      <Button type="default" onClick={() => {}}>
        <p>a button</p>
      </Button>,
    );
    expect(component).toMatchSnapshot();
  });

  it('should call onClick on a click', () => {
    const mockClickEvent = { preventDefault };

    const component = mount(
      <Button type="default" onClick={onClickMock}>
        <p>a button</p>
      </Button>,
    );
    component.simulate('click', mockClickEvent);
    expect(onClickMock).toHaveBeenCalled();
  });
});
