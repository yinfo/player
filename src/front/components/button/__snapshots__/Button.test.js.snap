// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders as expected 1`] = `ShallowWrapper {}`;
